actions.js

```diff
@@ -55,3 +57,128 @@ export function constructContactItemsWithToken(token) {
     });
   };
 }
+
+export function getHeyMentorToken(token, authType) {
+  return async (dispatch) => {
+    dispatch({
+      type: 'SET_USER',
+      data: {
+        loading: true,
+      }
+    });
+    const response = await fetch(
+      `${API_URL}/register/${authType}?access_token=${token}`,
+      { method: 'post' }
+    );
+
+    try {
+      const responseJson = await response.json();
+      if (responseJson && !responseJson.error) {
+        dispatch({
+          type: 'SET_USER',
+          data: {
+            hmToken: responseJson,
+          }
+        });
+      }
+    } catch (err) {
+      dispatch({
+        type: 'SET_ERROR',
+        data: {
+          text: `${err}`,
+        }
+      });
+    }
+    dispatch({
+      type: 'SET_USER',
+      data: {
+        loading: false,
+      }
+    });
+  };
+}
+
+export function initFacebookLogin() {
+  return async (dispatch) => {
+    dispatch({
+      type: 'SET_USER',
+      data: {
+        loading: true,
+        loadingPlatform: 'facebook'
+      }
+    });
+    try {
+      const response = await Facebook.logInWithReadPermissionsAsync(
+        CONFIG.FACEBOOK_APP_ID,
+        {
+          permissions: ['public_profile', 'email', 'user_friends']
+        }
+      );
+      if (response.type === 'success') {
+        dispatch({
+          type: 'SET_USER',
+          data: {
+            fbToken: response.token,
+            gToken: null
```

Comment:
The reason I'm resetting `gToken` is that when the user has logged in via google and then facebook, the next time the app is launched it may automatically launch using google instead of facebook.

I don't mind either approach, if you'd prefer I remove them let me know

Reply:
Yeah I was thinking we should allow both the be set, but really, you probably shouldn't be able to see the login screen unless both tokens are invalid anyway, so it's fine to null it out.

Reply:
ah I see your point now, I removed it so we don't have unnecessary code